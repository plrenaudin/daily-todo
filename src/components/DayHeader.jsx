import React from "react";
import styled from "styled-components";
import { format } from "date-fns";

const Day = styled.td`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  padding: 1rem;
`;

const DayLabel = styled.em`
  font-size: 1.3rem;
  font-weight: bold;
`;

const DayDigit = styled.small`
  color: grey;
`;

const DayHeader = React.memo(({ day }) => {
  const [label, digit] = format(day, "dd Do").split(" ");
  return (
    <Day>
      <DayLabel>{label}</DayLabel>
      <DayDigit>{digit}</DayDigit>
    </Day>
  );
});

export { DayHeader };
