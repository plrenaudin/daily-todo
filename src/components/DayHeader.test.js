import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";

import { DayHeader } from "./DayHeader";

it("renders the DayHeader correctly", () => {
  const component = <DayHeader day={new Date(1549509463523)} />;
  expect(renderer.create(component).toJSON()).toMatchSnapshot();
});
