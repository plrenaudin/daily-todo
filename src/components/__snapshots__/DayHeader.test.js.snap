// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the DayHeader correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 1rem;
}

.c1 {
  font-size: 1.3rem;
  font-weight: bold;
}

.c2 {
  color: grey;
}

<td
  className="c0"
>
  <em
    className="c1"
  >
    We
  </em>
  <small
    className="c2"
  >
    6th
  </small>
</td>
`;
